### PR TITLE
Fix retains for embedded types in YTextEvent

### PR DESCRIPTION
**Pull request by ezeidman**
Don't include attributes in retains for embedded types since they're not subject to formatting

This fixes https://github.com/yjs/yjs/issues/529

### CODE REVIEW

**Comment by dmonad** [comment]
Thank you! 
Can you please confirm that this is the same behavior as in https://github.com/quilljs/delta/ ?

**Comment by ezeidman** [comment]
> Thank you! Can you please confirm that this is the same behavior as in https://github.com/quilljs/delta/ ?

Could you clarify what you mean? The deltas from both before and after the PR are valid Quill deltas. The issue however is that retains associated with embedded types (e.g. an XmlText inside an XmlText) can contain unrelated attribute updates that are actually associated with surrounding text. This could be an erroneous null attribute like the example in the linked ticket or an erroneous non-null attribute value. 

An example:

```
  const doc =  new Y.Doc()
  const root = doc.get("text", Y.XmlText) as Y.XmlText;
  const p1 = new Y.XmlText();
  p1.setAttribute("type", "paragraph")
  p1.insert(0, "p1")
  
  root.insert(0, "abc", {bold: true});
  root.insert(3, "defghi", {bold: false});
  root.insertEmbed(6, p1);
  
  root.observeDeep((events) => {
    events.forEach(event => {
      console.log("Event", JSON.stringify(event.changes, null, 4));
    })
  })

  root.delete(3, 3);
```

produces
Screenshot 2023-05-05 at 2 28 00 PM

where the retain is applying bold=false to the XmlText even though it's not bold and neither are its contents.

There is one issue with this PR as is: it can create more retains than are strictly necessary. For example [retain 3, retain 4, retain 1], instead of just retain 8. That's still a valid Quill delta but not the most concise way to represent things.

File: tests/y-text.tests.js [hunk 1]
```diff
@@ -1673,6 +1673,82 @@ export const testDeltaAfterConcurrentFormatting = tc => {
...
+   */
+  const deltas = []
+  text1.observe(event => {
+    if (event.delta.length > 0) {
+      deltas.push(event.delta)
+    }
+  })
+  text0.delete(3, 3);
+  testConnector.flushAllMessages()
+  t.compare(deltas, [[{retain: 3}, { delete: 3}]])
```
Comment (ezeidman) [inline tests/y-text.tests.js:1727]: Before the fix this is `[[{"retain":3},{"delete":3},{"retain":4,"attributes":{"bold":false}}]]`

File: tests/y-text.tests.js [hunk 2]
```diff
@@ -1673,6 +1673,82 @@ export const testDeltaAfterConcurrentFormatting = tc => {
...
+   */
+  const deltas = []
+  text1.observe(event => {
+    if (event.delta.length > 0) {
+      deltas.push(event.delta)
+    }
+  })
+  text0.delete(0, 3);
+  testConnector.flushAllMessages()
+  t.compare(deltas, [[{ delete: 3}]])
```
Comment (ezeidman) [inline tests/y-text.tests.js:1700]: Before the fix this is `[[{"delete":3},{"retain":1,"attributes":{"bold":null}}]]`

**Pull request closed without merging**